streams/src/test/java/org/apache/kafka/streams/integration/KTableKTableForeignKeyInnerJoinMultiIntegrationTest.java

```diff
@@ -172,7 +174,7 @@ public void after() throws IOException {
             streamsThree.close();
             streamsThree = null;
         }
-        IntegrationTestUtils.purgeLocalStreamsState(streamsConfig);
+        IntegrationTestUtils.purgeLocalStreamsState(asList(streamsConfig, streamsConfigTwo, streamsConfigThree));
```

Comment:
another fix: we missed it in this fix: https://github.com/apache/kafka/pull/9978/files#diff-29da306477a4e2aaf1195f085dd77ae150f767e96277db52252d4093f802a9aeR154, to separate the streamConfig to persist UUID in state directory.

Reply:
This is done above in `before()` so it's probably not behind the flakiness, unfortunately. Not sure why we purge the local state twice but you're right, we should at least be consistent when we do and purge the state for all three Streams. Good catch